### PR TITLE
Create sonarcloud.yml

### PR DESCRIPTION
**Pull request by geeknoid**
Enable SonarCloud CI step.

### CODE REVIEW

File: .github/workflows/sonarcloud.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        uses: SonarSource/sonarcloud-github-action@de2e56b42aa84d0b1c5b622644ac17e505c9a049
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}   # Generate a token on Sonarcloud.io, add it to the secrets of this repo with the name SONAR_TOKEN (Settings > Secrets > Actions > add new repository secret)
+        with:
+          # Additional arguments for the sonarcloud scanner
+          args:
+            # Unique keys of your project and organization. You can find them in SonarCloud > Information (bottom-left menu)
+            # mandatory
+            -Dsonar.projectKey=
```
Comment (tcdoan) [inline .github/workflows/sonarcloud.yml:59]: Do we need to supply this argument?

**Review by tcdoan — APPROVED** [review]

File: .github/workflows/sonarcloud.yml (hunk 1 above)
Comment (geeknoid) [reply .github/workflows/sonarcloud.yml:59]: I thought so, but I wanted to try an experiment.

It does seem to have run without this information.

Comment (tcdoan) [reply .github/workflows/sonarcloud.yml:59]: It does look like so.

**Comment by tcdoan** [comment]
Do we know why ["SonarCloud analysis" failed](https://github.com/FRC-1294/frc2024/actions/runs/6133726959/job/16645819836?pr=93).

**Comment by tcdoan** [comment]
This **docker run** failed...

`/usr/bin/docker run --name ef7d85f55b2699128b4631b51113d00f6851f2_280d6b --label ef7d85 --workdir /github/workspace --rm -e "GITHUB_TOKEN" -e "SONAR_TOKEN" -e "INPUT_ARGS" -e "INPUT_PROJECTBASEDIR" -e "HOME" -e "GITHUB_JOB" -e "GITHUB_REF" -e "GITHUB_SHA" -e "GITHUB_REPOSITORY" -e "GITHUB_REPOSITORY_OWNER" -e "GITHUB_REPOSITORY_OWNER_ID" -e "GITHUB_RUN_ID" -e "GITHUB_RUN_NUMBER" -e "GITHUB_RETENTION_DAYS" -e "GITHUB_RUN_ATTEMPT" -e "GITHUB_REPOSITORY_ID" -e "GITHUB_ACTOR_ID" -e "GITHUB_ACTOR" -e "GITHUB_TRIGGERING_ACTOR" -e "GITHUB_WORKFLOW" -e "GITHUB_HEAD_REF" -e "GITHUB_BASE_REF" -e "GITHUB_EVENT_NAME" -e "GITHUB_SERVER_URL" -e "GITHUB_API_URL" -e "GITHUB_GRAPHQL_URL" -e "GITHUB_REF_NAME" -e "GITHUB_REF_PROTECTED" -e "GITHUB_REF_TYPE" -e "GITHUB_WORKFLOW_REF" -e "GITHUB_WORKFLOW_SHA" -e "GITHUB_WORKSPACE" -e "GITHUB_ACTION" -e "GITHUB_EVENT_PATH" -e "GITHUB_ACTION_REPOSITORY" -e "GITHUB_ACTION_REF" -e "GITHUB_PATH" -e "GITHUB_ENV" -e "GITHUB_STEP_SUMMARY" -e "GITHUB_STATE" -e "GITHUB_OUTPUT" -e "RUNNER_OS" -e "RUNNER_ARCH" -e "RUNNER_NAME" -e "RUNNER_ENVIRONMENT" -e "RUNNER_TOOL_CACHE" -e "RUNNER_TEMP" -e "RUNNER_WORKSPACE" -e "ACTIONS_RUNTIME_URL" -e "ACTIONS_RUNTIME_TOKEN" -e "ACTIONS_CACHE_URL" -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/frc2024/frc2024":"/github/workspace" ef7d85:f55b2699128b4631b51113d00f6851f2 -Dsonar.projectKey= -Dsonar.organization=`

File: .github/workflows/sonarcloud.yml (hunk 1 above)
Comment (tcdoan) [reply .github/workflows/sonarcloud.yml:59]: I looked at the [step 8 of the SonarCloud analysis docker run](https://github.com/FRC-1294/frc2024/actions/runs/6133726959/job/16645819836?pr=93). At the end of the command the last 2 args are empty.  

I looked at file [.github/workflows/sonarcloud.yml](https://github.com/FRC-1294/frc2024/pull/93/commits/ff7336c355d959cf496b78e1754854fa527c9603)

- Line 13 asked "Add our GitHub organization first, then add our repository as a new project."
- Line 19 asked "Copy/paste the Project Key and the Organization Key into the args parameter specified at line 59 "

Comment (tcdoan) [reply .github/workflows/sonarcloud.yml:59]: After that the "analysis is completed" but it failed. Looks like we do need to have 2 env variables **sonar.projectKey** and **sonar.organization** set

**Pull request closed without merging**